src/tbb/dynamic_link.cpp

```diff
@@ -387,7 +387,11 @@ namespace r1 {
     #endif /* !__TBB_DYNAMIC_LOAD_ENABLED */
         // RTLD_GLOBAL - to guarantee that old TBB will find the loaded library
         // RTLD_NOLOAD - not to load the library without the full path
+        #ifdef __HAIKU__
+        library_handle = dlopen(library, RTLD_LAZY | RTLD_GLOBAL);
```

Comment:
a small note on this one. RTLD_NOLOAD is not a standard POSIX define, so other platforms may benefit from this. (BSD's namely)